File: problems/3083-existence-of-a-substring-in-a-string-and-its-reverse/solution.py
class Solution:
    def isSubstringPresent(self, s: str) -> bool:
        n = len(s)
        if n < 2:
            return False
        reversed_s = s[::-1]
        for i in range(n - 1):
            substring = s[i:i+2]
            if substring in reversed_s:
                return True
        return False

File: problems/3083-existence-of-a-substring-in-a-string-and-its-reverse/solution.java
class Solution {
    public boolean isSubstringPresent(String s) {
        if (s.length() < 2) {
            return false;
        }

        String reversedS = new StringBuilder(s).reverse().toString();

        for (int i = 0; i < s.length() - 1; i++) {
            String sub = s.substring(i, i + 2);
            if (reversedS.contains(sub)) {
                return true;
            }
        }

        return false;
    }
}